Rename `rippled.cfg` to `xrpld.cfg`

bthomee opened the pull request:
## High Level Overview of Change

This change renames all occurrences of `rippled.cfg` to `xrpld.cfg`. It also provides a script to allow developers to replicate the changes in their local branch or fork to avoid conflicts. For the time being it maintains support for `rippled.cfg` as config file, if `xrpld.cfg` does not exist.

### Context of Change

Per [XLS-0095](https://xls.xrpl.org/xls/XLS-0095-rename-rippled-to-xrpld.html), we are taking steps to rename ripple(d) to xrpl(d).

This change essentially ensures that the config file will now be named `xrpld.cfg`. At the same time, everyone can keep using the config file named `rippled.cfg. The plan is to remove support for `rippled.cfg` in about 6 months to give everyone plenty of time to update the config file name.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

bthomee committed:
Renames RIPPLED_ and RIPPLE_ definitions to XRPL_

bthomee committed:
Reduce scope

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Expand scope

bthomee committed:
Add readme

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Add CI step to enforce renaming

bthomee committed:
Specify directory

bthomee committed:
Add paths to CI trigger

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Use gnu-sed on MacOS

bthomee committed:
refactor: Remove unnecessary copyright notices already covered by LICENSE.md

bthomee committed:
Add modified files

bthomee committed:
Fix typo in comment

bthomee committed:
refactor: Rename cmake files and definitions

bthomee committed:
Fix typo in README

bthomee committed:
Handle rerunning of the script

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'bthomee/rename_definitions' into bthomee/rename_copyright

bthomee committed:
Merge branch 'bthomee/rename_copyright' into bthomee/rename_cmake

bthomee committed:
Restore renaming of the .cfg file

bthomee committed:
Merge branch 'develop' into bthomee/rename_definitions

bthomee committed:
Merge branch 'bthomee/rename_definitions' into bthomee/rename_copyright

bthomee committed:
Merge branch 'bthomee/rename_copyright' into bthomee/rename_cmake

bthomee committed:
Add cmake script to CI check pipeline

bthomee committed:
Fix bad merge

bthomee committed:
Merge branch 'bthomee/rename_definitions' into bthomee/rename_copyright

bthomee committed:
Merge branch 'bthomee/rename_copyright' into bthomee/rename_cmake

bthomee committed:
Work around rename command not being available in CI images

bthomee committed:
Remove useless command

bthomee committed:
Fix script

bthomee committed:
refactor: Rename `rippled` binary to `xrpld` and add symlink

bthomee committed:
refactor: Rename `rippled.cfg` to `xrpld.cfg`

bthomee committed:
Use pushd and popd

bthomee committed:
Merge branch 'bthomee/rename_copyright' into bthomee/rename_cmake

bthomee committed:
Use pushd and popd

bthomee committed:
Use braces and quotes appropriately

bthomee committed:
Merge branch 'bthomee/rename_definitions' into bthomee/rename_copyright

bthomee committed:
Use braces and quotes appropriately

bthomee committed:
Merge branch 'bthomee/rename_copyright' into bthomee/rename_cmake

bthomee committed:
Use braces and quotes appropriately

bthomee committed:
Merge branch 'bthomee/rename_cmake' into bthomee/rename_binary

bthomee committed:
Use braces and quotes appropriately

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_config

bthomee committed:
Use braces and quotes appropriately

bthomee committed:
refactor: Rename `rippled` binary to `xrpld`

bthomee committed:
Fix comments

bthomee committed:
Reverse symlink

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_config

bthomee committed:
Merge branch 'develop' into bthomee/rename_binary

bthomee committed:
Restore changes to CI pipeline undone by other PR

bthomee committed:
refactor: Rename `ripple` namespace to `xrpl`

bthomee committed:
Merge branch 'develop' into bthomee/rename_binary

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_namespace

bthomee committed:
Merge branch 'bthomee/rename_namespace' into bthomee/rename_config

bthomee committed:
Merge branch 'develop' into bthomee/rename_binary

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_namespace

bthomee committed:
Merge branch 'bthomee/rename_namespace' into bthomee/rename_config

bthomee committed:
Support reading both xrpld.cfg and rippled.cfg

bthomee committed:
Merge branch 'develop' into bthomee/rename_binary

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_namespace

bthomee committed:
Merge branch 'bthomee/rename_namespace' into bthomee/rename_config

bthomee committed:
Update script to exclude support for legacy config file name

bthomee committed:
Improve variable naming

dangell7 commented:
Validator Tools PR: https://github.com/ripple/validator-keys-tool/pull/60

dangell7 commented:
Lots of `rippled` references in the xrpld.cfg. Should `path=/var/lib/rippled/db/nudb` be `path=/var/lib/xrpld/db/nudb`?

bthomee committed:
Merge branch 'develop' into bthomee/rename_binary

bthomee committed:
Rename rippled to xrpld in readmes

bthomee committed:
Modify script to restore git repo name

bthomee committed:
Merge branch 'bthomee/rename_binary' into bthomee/rename_namespace

bthomee committed:
Merge branch 'bthomee/rename_namespace' into bthomee/rename_config

bthomee committed:
Rename test suites

bthomee committed:
Special case for test suite

bthomee committed:
Merge branch 'bthomee/rename_namespace' into bthomee/rename_config

bthomee committed:
Rename inside .cfg

bthomee commented:
> Lots of `rippled` references in the xrpld.cfg. Should `path=/var/lib/rippled/db/nudb` be `path=/var/lib/xrpld/db/nudb`?

Thanks for pointing this out. I adjusted the script to also rename inside the config file. There are many other files with references too, but those can be done as a separate clean up PR afterwards.

Regarding the paths, I renamed those too in the example config. I did not touch the config that someone might have stored in the repo root for testing, since they might depend on the current paths.

bthomee committed:
Merge branch 'develop' into bthomee/rename_config

bthomee committed:
Merge branch 'develop' into bthomee/rename_config

bthomee committed:
Add tests for HOME and XDG_CONFIG_HOME

bthomee committed:
Improve test description comment, reduce variables

bthomee committed:
Merge branch 'develop' into bthomee/rename_config

bthomee committed:
Merge branch 'develop' into bthomee/rename_config

bthomee committed:
Merge branch 'develop' into bthomee/rename_config

legleux commented on src/xrpld/core/detail/Config.cpp:
```diff
@@ -374,11 +377,9 @@ Config::setup(
-            dataDir.clear();
     }
 
-    if (!dataDir.empty())
```
This is suspect

bthomee replied:
Yes - there was a reason why I thought this would be fine, but that must've been based on flawed assumptions. See https://github.com/XRPLF/rippled/pull/6301 for the fix.

legleux commented on src/xrpld/core/detail/Config.cpp:
```diff
@@ -295,76 +296,78 @@ Config::setup(
+            if (boost::filesystem::exists(CONFIG_FILE))
+                break;
+            CONFIG_FILE = CONFIG_DIR / configLegacyName;
+        } while (false);
```
ok?

bthomee replied:
Do you see a specific concern?
